pkg/R/utils.R

```diff
@@ -164,11 +176,14 @@ reserialize <- function(rdd) {
   if (!inherits(rdd, "RDD")) {
     stop("Argument 'rdd' is not an RDD type.")
   }
-  if (rdd@env$serialized) {
-    return(rdd)
-  } else {
+  if (rdd@env$serializedMode != "byte") {
+    if (rdd@env$serializedMode == "row") {
+      rdd@env$serialzedMode == "byte"
```

Comment:
Only in the case that you are combining a `row` RRDD with another serialization type (e.g. in `UnionRDD` or if you are writing it out with a function that needs it to be `byte` (e.g. `saveAsObjectFile`).

So:
- `lapply` on a `row` RRDD will return another `row` RRDD
- `UnionRDD` on two `row` RRDDs will return a `row` RRDD
- `UnionRDD` on a `row` RRDD and a `byte` or `string` RRDD will return a `byte` RRDD

The `byte` serialization mode is only forced in cases where we we need to call `reserialize`. Otherwise, `getJRDD` will still respect the incoming serialization type.